src/mobile/components/layout/bootstrap.coffee

```diff
@@ -59,7 +59,8 @@ ensureFreshUser = (data) ->
     'collector_level'
   ]
   for attr in attrs
-    if not _.isEqual data[attr], sd.CURRENT_USER[attr]
+    if (data[attr] or sd.CURRENT_USER[attr]) and not _.isEqual data[attr], sd.CURRENT_USER[attr]
```

Comment:
This additional condition guards against a `null` and `""` being compared and causing a refresh. I did notice that happening (if they're both false-y, we don't need to do a fuller equality check).

Reply:
I wonder if that was the original issue....